src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/IndexManagementPlugin.kt

```diff
@@ -240,6 +238,8 @@ internal class IndexManagementPlugin : JobSchedulerExtension, NetworkPlugin, Act
             .registerConsumers()
         rollupInterceptor = RollupInterceptor(clusterService, settings, indexNameExpressionResolver)
         this.indexNameExpressionResolver = indexNameExpressionResolver
+
+        val skipFlag = SkipExecution(client, clusterService)
```

Comment:
I think this might be useful in job scheduler as a generic solution for all plugins extending it to avoid running jobs in a cluster during upgrades. Can we add a TODO to the SkipExecution class to do this.

Reply:
added